addon/components/yeti-table/thead/row/column.js

```diff
@@ -56,50 +60,98 @@ class Column extends DidChangeAttrsComponent {
    *
    * If you don't need sorting, filtering or automatic table unrolling (using the blockless
    * body component), then this property is optional.
+   *
+   * @argument prop
+   * @type {String}
    */
-  prop;
+  @dependentKeyCompat
+  get prop() {
+    return this.args.prop;
+  }
 
+  @tracked
+  _visible;
   /**
    * Set to `false` to hide the entire column across all rows. Keep in mind that this property
    * won't just hide the column using css. The DOM for the column will be removed. Defaults to `true`.
+   *
+   * @argument visible
+   * @type {Boolean}
    */
-  visible = true;
+  @dependentKeyCompat
+  get visible() {
+    return this._visible === undefined ? this.getArgWithDefault('visible', true) : this._visible;
+  }
+  set visible(value) {
+    this._visible = value;
+  }
```

Comment:
@cah-briangantzler are you aware about the `@localCopy` decorator of the https://github.com/pzuraq/tracked-toolbox project?

I think this is a perfect fit for these situations and get a rid of some boilerplate.